File: web/src/api/campusCategory.js
import service from '@/utils/request'

export const getCampusCategoryList = (params) => {
  return service({
    url: '/campusCategory/getCampusCategoryList',
    method: 'get',
    params
  })
}

export const findCampusCategory = (params) => {
  return service({
    url: '/campusCategory/findCampusCategory',
    method: 'get',
    params
  })
}

export const createCampusCategory = (data) => {
  return service({
    url: '/campusCategory/createCampusCategory',
    method: 'post',
    data
  })
}

export const updateCampusCategory = (data) => {
  return service({
    url: '/campusCategory/updateCampusCategory',
    method: 'put',
    data
  })
}

export const updateCampusCategoryStatus = (data) => {
  return service({
    url: '/campusCategory/updateCampusCategoryStatus',
    method: 'post',
    data
  })
}

File: server/router/campus/campus_category.go
package campus

import (
	"github.com/flipped-aurora/gin-vue-admin/server/middleware"
	"github.com/gin-gonic/gin"
)

type CampusCategoryRouter struct{}

func (r *CampusCategoryRouter) InitCampusCategoryRouter(Router *gin.RouterGroup) {
	categoryRouter := Router.Group("campusCategory").Use(middleware.OperationRecord())
	categoryRouterWithoutRecord := Router.Group("campusCategory")
	{
		categoryRouter.POST("createCampusCategory", campusCategoryApi.CreateCampusCategory)
		categoryRouter.PUT("updateCampusCategory", campusCategoryApi.UpdateCampusCategory)
		categoryRouter.POST("updateCampusCategoryStatus", campusCategoryApi.UpdateCampusCategoryStatus)
	}
	{
		categoryRouterWithoutRecord.GET("getCampusCategoryList", campusCategoryApi.GetCampusCategoryList)
		categoryRouterWithoutRecord.GET("findCampusCategory", campusCategoryApi.FindCampusCategory)
	}
}
